cmd/pebble/test.go

```diff
@@ -324,27 +326,33 @@ func runTest(dir string, t test) {
 	workersDone := make(chan struct{})
 	signal.Notify(done, os.Interrupt)
 
-	go func() {
+	eg, ctx := errgroup.WithContext(context.Background())
+	defer ctx.Done()
+
+	eg.Go(func() error {
 		wg.Wait()
 		close(workersDone)
-	}()
+		return nil
+	})
 
 	if maxSize > 0 {
-		go func() {
+		eg.Go(func() error {
 			for {
 				time.Sleep(10 * time.Second)
 				if db.Metrics().DiskSpaceUsage() > maxSize*1e6 {
```

Comment:
You'd need a way to exit this loop if the context is cancelled. You can do it with a simple:

```
if err := ctx.Err(); err != nil {
  return nil
}
```